### PR TITLE
add aggregated ClusterRole to manage namespaced resources

### LINKED ISSUE #1398: user roles for namespaced resources are missing in logging-operator helm chart
**Describe the bug**:

A user should have access to namespaced resources like Flows and Outputs. It seems this is not covered by the logging-operator Helm chart,only a global ClusterRole with ClusterRoleBinding will created. 

**Expected behaviour**:
User with admin permissions for a namespace should have access to Flows and Outputs of this namespace.

**Steps to reproduce the bug**:
Try to get access to an existing Output:

```
 outputs.logging.banzaicloud.io "demoapp" is forbidden: User "u-xxxxxx" cannot get resource "outputs" in API group "logging.banzaicloud.io" in the namespace "demoapp"
```

/kind bug

### PR DESCRIPTION
**Pull request by eumel8**
fixes #1398

### CODE REVIEW

File: charts/logging-operator/templates/userrole.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
+rules:
+- apiGroups:
+  - logging-extensions.banzaicloud.io
+  resources:
+  - eventtailers
+  verbs:
+  - create
+  - delete
```
Comment (aslafy-z) [inline charts/logging-operator/templates/userrole.yaml:16]: Please add `deletecollection` to all resources

File: charts/logging-operator/templates/userrole.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  - update
+  - watch
+- apiGroups:
+  - logging.banzaicloud.io
+  resources:
+  - syslogngflows/status
+  - syslogngoutputs/status
+  verbs:
+  - get
+  - patch
```
Comment (aslafy-z) [inline charts/logging-operator/templates/userrole.yaml:72]: Is `patch` and `update` verbs needed in here?

Comment (eumel8) [reply charts/logging-operator/templates/userrole.yaml:75]: good question. would that be an effect if a user updates the status manually?

Comment (pepov) [reply charts/logging-operator/templates/userrole.yaml:75]: the status is patched/updated by the operator exclusively, not something for the user to do.

**Comment by pepov** [comment]
Thanks for the proposal and for the implementation! However I would like to separate these clusterroles based on the following resources:
- host tailer
- flows & outputs + syslogngflows & outputs
- clusterflows & outputs + syslogngclusterflows & outputs

I'm not sure about the eventtailer, since that is something to be installed once into a logging system, not something to manage by individual users.

I would add the edit/admin aggregation labels, so that it can easily be aggregated into a single clusterrole for editor/admin purposes, like it is done in the cert-manager chart:
https://github.com/cert-manager/cert-manager/blob/0b9366c0fb226393f2c51abdf9ec3097606a8713/deploy/charts/cert-manager/templates/rbac.yaml#L439

Also I would create a viewer role, that doesn't have to be split by resources, just grant view privileges to all the above logging resources for users.

I'm still a bit sceptical that we can provide clusterroles that are useful for most users/use cases, but at least we can provide an example implementation.

File: charts/logging-operator/templates/userrole.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+kind: ClusterRole
+metadata:
+  name: {{ template "logging-operator.fullname" . }}-user
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
+rules:
+- apiGroups:
+  - logging-extensions.banzaicloud.io
+  resources:
+  - eventtailers
```
Comment (pepov) [inline charts/logging-operator/templates/userrole.yaml:13]: eventtailer is not a namespaced resource, I don't think we should add it here

File: charts/logging-operator/templates/userrole.yaml (hunk 2 above)
Comment (eumel8) [reply charts/logging-operator/templates/userrole.yaml]: ok, removed

**Comment by eumel8** [comment]
@pepov Rancher has this implementation also for [read-only/view access](https://github.com/rancher/charts/blob/release-v2.7/charts/rancher-logging/102.0.1%2Bup3.17.10/templates/userroles.yaml). 
Aggregation is a common RBAC feature in Kubernetes and useful not only in context with Rancher projects. An extra value flag under the rbac section of the chart is also possible. At the moment normal user haven't  access to Flow and Output resources.

File: charts/logging-operator/templates/userrole.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+{{- if .Values.rbac.enabled }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "logging-operator.fullname" . }}-user
```
Comment (pepov) [inline charts/logging-operator/templates/userrole.yaml:6]:
```suggestion
  name: {{ template "logging-operator.fullname" . }}-edit
```

File: charts/logging-operator/templates/userrole.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
```
Comment (pepov) [inline charts/logging-operator/templates/userrole.yaml:8]:
```suggestion
    rbac.authorization.k8s.io/aggregate-to-admin: "true"
    rbac.authorization.k8s.io/aggregate-to-edit: "true"
```

File: charts/logging-operator/templates/userrole.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - logging.banzaicloud.io
+  resources:
+  - syslogngflows/status
+  - syslogngoutputs/status
+  verbs:
+  - get
```
Comment (aslafy-z) [inline charts/logging-operator/templates/userrole.yaml:51]: Is `watch` a verb that applies to status subresources?

File: charts/logging-operator/templates/userrole.yaml (hunk 5 above)
Comment (aslafy-z) [reply charts/logging-operator/templates/userrole.yaml:8]: Looks like the edit ClusterRole is already aggregated to admin, I guess adding `aggregate-to-edit` here should be sufficient

File: charts/logging-operator/templates/userrole.yaml (hunk 6 above)
Comment (pepov) [reply charts/logging-operator/templates/userrole.yaml:51]: I don't think so. Also I'm not entirely sure, why we need get on status. Do we need that at all?

Comment (eumel8) [reply charts/logging-operator/templates/userrole.yaml:52]: You mean the syslogng resource or all resources in general? Otherweise the user can't get access to the status.

Comment (pepov) [reply charts/logging-operator/templates/userrole.yaml:52]: I mean if you have get access on a resource, you should have get access on the subresource as well.

Comment (eumel8) [reply charts/logging-operator/templates/userrole.yaml]: Yes, make sense. Tested and the status is editable in any way, but the Operator sets back.

**Review by pepov — APPROVED** [review]

**Pull request merged**